frontend/javascripts/oxalis/model.js

```diff
@@ -148,10 +202,22 @@ export class OxalisModel {
   }
 
   getPullQueueByLayerName(name: string): PullQueue {
+    return this.getLayerByName(name).pullQueue;
+  }
+
+  getLayerByName(name: string): DataLayer {
     if (!this.dataLayers[name]) {
       throw new Error(`Layer with name ${name} was not found.`);
     }
-    return this.dataLayers[name].pullQueue;
+    return this.dataLayers[name];
+  }
+
+  getSegmentationLayerWithMappingSupport(): ?DataLayer {
+    const layer = getSegmentationLayerWithMappingSupport(Store.getState());
```

Comment:
Would it make sense to use different names indicating the difference between the two functions?

Reply:
Hmm, maybe.. Not sure if this makes things easier. The difference between `DataLayer` and `APIDataLayer` is quite annoying in general (often, when having a `layer` variable, one has to find out first what exactly it is). Usually, I spot the difference between the two `getSegmentationLayerWithMappingSupport` (or similar) functions by their usage (either it's a method on `Model` or it's a function which consumes a `state`). In the long term, I want to get rid of the duplication (not sure how, though). Maybe having the same function name twice is a good indicator of this smell?

Reply:
That's a good description of the issue :+1: We'll likely end up with some combination of renamings. The `APIDataLayer` could be `LayerSpec` or `LayerConfig`. The `DataLayer` could keep its name or be `LayerInstance`. However, I don't know all the details, so there might be a better solution.
I agree with leaving this as-is to keep the smell :grin: